corehq/motech/auth.py

```diff
@@ -39,3 +105,153 @@ def __call__(self, r):
         token = self._get_auth_token(r)
         r.headers["Authorization"] = f"Bearer {token}"
         return r
+
+
+class AuthManager:
+
+    def get_auth(self) -> Optional[AuthBase]:
+        """
+        Returns an instance of requests.auth.AuthBase, to be passed to
+        an outgoing API request, or None if not applicable.
+        """
+        return None
+
+    def get_session(self) -> Session:
+        """
+        Returns an instance of requests.Session. Manages authentication
+        tokens, if applicable.
+        """
+        session = Session()
+        session.auth = self.get_auth()
+        return session
+
+
+class BasicAuthManager(AuthManager):
+
+    def __init__(self, username, password):
+        self.username = username
+        self.password = password
+
+    def get_auth(self):
+        return HTTPBasicAuth(self.username, self.password)
+
+
+class DigestAuthManager(AuthManager):
+
+    def __init__(self, username, password):
+        self.username = username
+        self.password = password
+
+    def get_auth(self):
+        return HTTPDigestAuth(self.username, self.password)
+
+
+class OAuth1Manager(AuthManager):
+
+    def __init__(
+        self,
+        client_id: str,
+        client_secret: str,
+        api_endpoints: OAuth1ApiEndpoints,
+        last_token: Optional[dict] = None,
+    ):
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.api_endpoints = api_endpoints
```

Comment:
looks like this isn't used (yet)

Reply:
Yeah, I don't know at what point we started claiming to support OAuth1, but I don't think we ever have. I was unsure whether to just drop "OAuth1" as an authentication type for Repeaters, or whether to put in place the things we'd need to support it if anyone ever calls us on our claim.

(I could easily be persuaded to drop it and pretend nobody ever mentioned it. :zipper_mouth_face: )